pkg/watcher/skr_webhook_manifest_manager.go

```diff
@@ -87,9 +85,13 @@ func (m *SKRWebhookManifestManager) Install(ctx context.Context, kyma *v1beta2.K
 		return fmt.Errorf("failed to get skrContext: %w", err)
 	}
 
+	if _, err = gatewaysecret.GetGatewaySecret(ctx, m.kcpClient); err != nil {
```

Comment:
Isn't that redundant? Later on, this value is required anyway so we make another call like this.

Reply:
Yes, it was a part of the issue that the watcher secret is only created if the gateway secret is ready.

Later on, we fetch the gateway secret to check if the watcher secret needs to be updated; if an error occurs there, we don't update the watcher secret, but it will be created.
Therefore, stopping the watcher's secret from being created altogether when the gateway is not ready makes sense and is not redundant.

Reply:
```Later on, we fetch the gateway secret to check if the watcher secret needs to be updated;```
I still don't get it. We fetch the secret in two places in this file: line 88 and 235. This places are quite "close" to each other with respect to actual execution. Can't we use the object fetched in line 88 also in line 235?
I mean, if there is a justified likelyhood that this secret somehow changes between lines 88 and 235, then it's fine to fetch it again.
Otherwise it is a code smell: Somehow the current code structure makes it hard for us to fetch the object just once and then use it.
If you don't know how to fix it in the current codebase (Or you don't want to do it in this PR), I'd suggest to create an issue dedicated to refactoring of this "double lookup".